### DISCUSSION TITLE
Discord as a comunication server

### DISCUSSION THREAD

**Discussion by Sasanidas** [emacs-lsp/lsp-mode#2894] [Ideas]
Hello lsp-mode mantainers,
 I saw that recently that the project start using discord as a way to communicate internally and with other people.

 Been that lsp-mode is a free software package that is use by GNU Emacs, I'm interesting in know why Discord was selected as main way of communication, as far as I know, Discord [is not a friendly platform for free software ](https://sneak.berlin/20200220/discord-is-not-an-acceptable-choice-for-free-software-projects/) and I think moving the community to a centralize proprietary platform is not ideal for a free software project.

I willing to help to find an alternative, even looking for fund, if the main developers agree to it.

Regards

**Comment by ericdallo** [comment]
Hello @Sasanidas!
That was an idea to use a community channel more friendly than Gitter since the notifications didn't work for some users, I agree a free software would be good, but I don't know any other as good as Discord where we can have multiple channels, decent threads/replies and things like that.
Suggestions are welcome too :)

**Comment by brotzeit** [comment]
I can absolutely understand this concern. Actually, the gitter room is still active though it isn't linked from the github page anymore. But for my taste it's already enough that we use github. However if people decide that they prefer discord over gitter maybe we should also respect this decision. Since we already have two active channels, we won't stop people from communicating there unless we delete them. I think that would also be a little rude.

I would consider the option to link both channels and let the user decide. In my experience you can't convince people to change their mind about these kind of things.

**Comment by yyoncho** [comment]
I second @brotzeit. Let's restore the gitter room link and let the users decide what they want to use.

**Comment by Sasanidas** [comment]
> I can absolutely understand this concern. Actually, the gitter room is still active though it isn't linked from the github page anymore. But for my taste it's already enough that we use github. However if people decide that they prefer discord over gitter maybe we should also respect this decision. Since we already have two active channels, we won't stop people from communicating there unless we delete them. I think that would also be a little rude.
> 
> I would consider the option to link both channels and let the user decide. In my experience you can't convince people to change their mind about these kind of things.

I can totally understand that Gitter doesn't offer the same level of functionality as Discord, so offering both options may separate the community, and maybe doesn't work as intended.

I think that looking for a free software alternative to Discord that has similar functionalities maybe better to attract people that came from Discord.

From the top of my head, [Matrix](https://matrix.org/) and [Mattermost](https://mattermost.com/)

**Comment by folliehiyuki** [comment]
As a regular Matrix user I second this platform as well. You can easily bridge Discord and Gitter to Matrix too, so it will be easier to manage multiple channels in one place

**Comment by yyoncho** [comment]
I think that bridging discord and gitter to matrix is a good idea and it seems like it will make everyone happy. AFAIK GNU guidelines are not against proprietary services as long as they don't force the user to use proprietary software. In our case, briding to matrix will fill that gap.

**Comment by ericdallo** [comment]
Yeah, looks a good alternative indeed

**Comment by eras** [comment]
It seems it should be mentioned that the gitter room is already available in Matrix via the official Matrix-Gitter bridge which was implemented after Gitter was acquired by Element, which is a driving force behind Matrix: https://matrix.to/#/#emacs-lsp_lsp-mode:gitter.im

**Comment by brotzeit** [comment]
But the discord bridge doesn't work yet ? And how does this work with multiple rooms, do all discord rooms get mirrored to matrix ?

**Comment by eras** [comment]
There is a discord bridge project, but such bridge installations are not generally available for all, but instead need to be self-hosted. However, this can be done with free resources such as Heroku. I don't have experience on the service.

But all Gitter rooms are available over Matrix automatically as the people behind Matrix are now basically the same as with Gitter. Indeed this bridge is probably able to eventually handle everything as smoothly as possible given they are in control of both sides of the system, while Discord bridge can only work as good as the Discord API allows for.

The Discord bridge project can be found here: https://github.com/git-bruh/matrix-discord-bridge/

**Comment by xendk** [comment]
I'm a bit late to the party, but couldn't let this pass without throwing in a suggestion: [Zulip](https://zulip.com/).

It's Open Source (https://github.com/zulip/zulip/) and [offers free hosting for Open Source projects](https://zulip.com/for/open-source/)

Why Zulip? In addition to pointing to [their own reasoning](https://zulip.com/why-zulip/), I'll add my personal opinion: Zulips thread handling makes following and re-finding conversations much easier and is, in my opinion, much better suited for OSS projects where contributors are online at different times than traditional chat systems. Having multiple conversations at different paces is a breeze, and having conversations organized in topics means one can mute those they don't need to participate in. And the keyboard controls is right up an Emacs users ally (I only use the mouse for pressing play in embedded videos and clicking links).

At work we chose Zulip over the alternatives mentioned in this thread, and haven't looked back.

**Comment by ghost** [comment]
I'm a little confused, should I be using github discussions, the gitter/matrix room or the discord?

**Reply by ericdallo** [reply]
We recommend Discord or gitter, we'll answer on any of them
